### PR TITLE
[Keymap] CRKBD Custom Keymap - KidBrazil

### PR DESCRIPTION
**Pull request by kidBrazil**
## Description
Simple custom keymap for the CRKBD suited to fit most programmer/gamer types. Includes a toggle layer dedicated to gaming that closely approximates the mapping in most FPS games.
- Includes custom font for Logo
- Oled has been customized to show layer and USB state
- Oled switches to GAMING mode
- Gaming layer has one RAISE layer for weapon selection

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/kidbrazil/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#KidBrazil's custom CRKBD Layout
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/README.md]:
```suggestion
# KidBrazil's custom CRKBD Layout
```
GitHub's Markdown renderer requires a space here.

File: keyboards/crkbd/keymaps/kidbrazil/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Custom Font
+This keymap includes a custom font for my LooseTransistor logo. It is fine if
+you want to use it I certainly don't mind but if you want your own, just replace
+the glcdfont file here and you should be fine.
+
+Alternatively you could remove the font config line from the config.h file so it
+will use the default QMK one.
+
+##OLED
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/README.md]:
```suggestion
## OLED
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    _WEAPON
+};
+
+// [Keymaps] -----------------------------------------------------------------//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	  [_QWERTY] = LAYOUT(\
+      KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T,                                  KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC,\
+      LSFT_T(KC_TAB), KC_A, KC_S, KC_D, KC_F, KC_G,                          KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,\
+      KC_LCTL, KC_Z, KC_X, KC_C, KC_V, KC_B,                                 KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,\
+                          LGUI_T(KC_PGUP), MO(_NUM), KC_SPC,         KC_ENT, MO(_SYM), LALT_T(KC_PGDN)\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/crkbd/keymaps/kidbrazil/rules.mk [hunk 4]
```diff
@@ -0,0 +1,17 @@
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/rules.mk]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+};
+
+//int RGB_current_mode;
+
+// [Persistent Default Layer] ------------------------------------------------//
+void persistent_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

And it looks like you're not even using it.

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+// Render Logo
+void render_logo(void) {
+    oled_write(read_logo(), false);
+}
+
+// Master OLED Screen (Left Hand )
+void render_master_oled(void) {
+    // Switch display based on Layer
+    switch (biton32(layer_state)){
```
Comment (drashna) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
    switch (get_highest_layer(layer_state)){
```

**Comment by kidBrazil** [comment]
Thanks @drashna and @fauxpark for the review, I have made the changes as suggested.

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+            break;
+        default:
+            oled_write_ln_P(PSTR("++++++++++++++++++++"), false);
+    }
+}
+
+// Render current layer state
+void render_layer_state(void){
+	// If you want to change the display of OLED, you need to change here
+    switch (biton32(layer_state)){
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
    switch (get_highest_layer(layer_state)){
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+// {OLED helpers} -----------------------------------------------//
+
+// Render Blank Space
+void render_space(void) {
+    oled_write_ln_P(PSTR("     "), false);
+}
+
+// Render separator lines for oled display
+void render_separator(void) {
+    switch (biton32(layer_state)){
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
    switch (get_highest_layer(layer_state)){
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    _QWERTY,
+    _NUM,
+    _SYM,
+    _GAME,
+    _WEAPON
+};
+
+// [Keymaps] -----------------------------------------------------------------//
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	  [_QWERTY] = LAYOUT(\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
	  [_QWERTY] = LAYOUT(
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+      KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T,                                  KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC,
+      LSFT_T(KC_TAB), KC_A, KC_S, KC_D, KC_F, KC_G,                          KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,
+      KC_LCTL, KC_Z, KC_X, KC_C, KC_V, KC_B,                                 KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,
+                          LGUI_T(KC_PGUP), MO(_NUM), KC_SPC,         KC_ENT, MO(_SYM), LALT_T(KC_PGDN)
+    ),
+
+	  [_NUM] = LAYOUT(\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
	  [_NUM] = LAYOUT(
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+      KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5,                                      KC_6, KC_7, KC_8, KC_9, KC_0, KC_DEL,
+      LSFT_T(KC_TAB), KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,                         KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, KC_F12, KC_NO,
+      KC_LCTL, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10,                               KC_PGUP, KC_PGDN, KC_HOME, KC_END, KC_F11, KC_NO,
+                               LGUI_T(KC_PGUP), KC_TRNS, KC_SPC,         KC_ENT, KC_TRNS, LALT_T(KC_PGDN)
+    ),
+
+	  [_SYM] = LAYOUT(\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
	  [_SYM] = LAYOUT(
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+      KC_ESC, KC_EXLM, KC_AT, KC_HASH, KC_DLR, KC_PERC,                                   KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
+      LSFT_T(KC_TAB), RGB_TOG, RGB_MOD, RGB_HUI, RGB_SAI, RGB_VAI,                        KC_MINS, KC_EQL, KC_LCBR, KC_RCBR, KC_PIPE, KC_GRV,
+      KC_LCTL, RGB_VAD, RGB_RMOD, RGB_HUD, RGB_SAD, TG(_GAME),                            KC_UNDS, KC_PLUS, KC_LBRC, KC_RBRC, KC_BSLS, KC_TILD,
+                                          LGUI_T(KC_PGUP), KC_TRNS, KC_SPC,       KC_ENT, KC_TRNS, LALT_T(KC_PGDN)
+    ),
+
+	  [_GAME] = LAYOUT(\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
	  [_GAME] = LAYOUT(
```

File: keyboards/crkbd/keymaps/kidbrazil/keymap.c [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+      KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T,                                   KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6,
+      KC_LSFT, KC_A, KC_S, KC_D, KC_F, KC_G,                                  KC_F7, KC_F8, KC_F9, KC_F10, KC_NO, KC_NO,
+      KC_LCTL, KC_Z, KC_X, KC_C, KC_V, KC_TRNS,                               KC_PGUP, KC_PGDN, KC_NO, KC_NO, KC_NO, KC_NO,
+                            KC_TAB, MO(_WEAPON), KC_SPC,              KC_ENT, KC_TRNS, KC_NO
+      ),
+
+	  [_WEAPON] = LAYOUT(\
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/kidbrazil/keymap.c]:
```suggestion
	  [_WEAPON] = LAYOUT(
```

**Pull request merged**